File: Solutions/297/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var x = GetInputs<int>(1).Single();
            var n = GetInputs<int>(1).Single();

            var sum = 1.0;
            for (var i = 1; i < n; i++)
                sum += Math.Pow(x, i) / Factorial(i);
            
            Console.WriteLine(sum.ToString("0.000"));
        }

        private static double Factorial(int number) {
            var result = 1.0;
            for (var i = number; i > 1; i--)
                result *= i;
            return result;
        }

        private static List<T> GetInputs<T>(int count) {
            var result = new List<T>(count);
            for (var i = 0; i < count; i++) {
                result.Add((T) Convert.ChangeType(Console.ReadLine(), typeof(T)));
            }

            return result;
        }
    }
}

File: Solutions/297/c++/program.cpp
// 2019-02-11

#include <iostream>
#include <iomanip>
#include <math.h>

using namespace std;

//Define:

//Prototypes:
double exponential(double, int);
long long factorial(int n);

int main()
{
	int x, n;
	cin >> x >> n;

	auto result = exponential(x, n);
	cout << fixed << std::setprecision(3) << result;
	return 0;
}


double exponential(double x, int n)
{
	if (x == 0)
		return 1;

	double sum = 1;
	for (int i = 0; i < n - 1; i++)
		sum += pow(x, i + 1) / factorial(i + 1);
	return sum;
}

long long factorial(int n)
{
	long long result = 1;
	while (n > 1)
	{
		result *= n;
		n--;
	}
	return result;
}
